### PR TITLE
Non-latin text-recognition support

### LINKED ISSUE #27: GoogleMLKit/TextRecognition pods can be updated to 3.2.0
According [Google Dev docs, Text Recognition v2 is here now.](https://developers.google.com/ml-kit/vision/text-recognition/v2/ios#before_you_begin):

> Include the following ML Kit pods in your Podfile:
>
> ```
> # To recognize Latin script
> pod 'GoogleMLKit/TextRecognition', '3.2.0'
> # To recognize Chinese script
> pod 'GoogleMLKit/TextRecognitionChinese', '3.2.0'
> # To recognize Devanagari script
> pod 'GoogleMLKit/TextRecognitionDevanagari', '3.2.0'
> # To recognize Japanese script
> pod 'GoogleMLKit/TextRecognitionJapanese', '3.2.0'
> # To recognize Korean script
> pod 'GoogleMLKit/TextRecognitionKorean', '3.2.0'
> ```
>
> After you install or update your project's Pods, open your Xcode project using its .xcworkspace. ML Kit is supported in Xcode version 12.4 or greater.

### PR DESCRIPTION
**Pull request by spetrey**
Closes #27

## What does this PR do?

This PR adds official text-recognition support for:
```
GoogleMLKit/TextRecognitionChinese
GoogleMLKit/TextRecognitionDevanagari
GoogleMLKit/TextRecognitionJapanese
GoogleMLKit/TextRecognitionKorean
```

## How to use

```tsx
const result = await TextRecognition.recognize(imagePath) // Latin text-recognition
...
const result = await TextRecognition.recognize(imagePath, 'Chinese') // Chinese text-recognition
...
const result = await TextRecognition.recognize(imagePath, 'Devanagari') // Devanagari text-recognition
...
const result = await TextRecognition.recognize(imagePath, 'Japanese') // Japanese text-recognition
...
const result = await TextRecognition.recognize(imagePath, 'Korean') // Korean text-recognition
```

## Results:

**Japanese and Devanagari**
Screenshot 2023-04-04 at 7 36 41 PM

**Chinese**
Screenshot 2023-04-04 at 7 34 38 PM

**Korean**
Screenshot 2023-04-04 at 7 33 41 PM

This is my first PR here! I will be the first to admit that Objective-C is not my strongest language, so I welcome any and all feedback! I would love for someone to pull this branch down and take it for a spin. My initial testing yielded some good results but would love someone to double-check my work and give me a sanity check 😄

### CODE REVIEW

File: text-recognition/index.d.ts [hunk 1]
```diff
@@ -62,9 +62,17 @@ interface ITextRecognition {
...
+   * text recognition. The default is English. Available languages:
+   * - Chinese
+   * - Devanagari
+   * - Japanese
+   * - Korean
+   *
    * @returns Text recognition result
    */
-  recognize: (imageURL: string) => Promise<TextRecognitionResult>;
+  recognize: (imageURL: string, language?: string | null) => Promise<TextRecognitionResult>;
```
Comment (a7medev) [inline text-recognition/index.d.ts:75]: I'd say let's make `language` an enum instead. See [@react-native-ml-kit/translate-text's `TranslateLanguage` enum](https://github.com/a7medev/react-native-ml-kit/blob/62ad4f3db2b083b8f527c36212ea7cc65a410cb8/translate-text/index.js#L9).

File: text-recognition/RNMLKitTextRecognition.podspec [hunk 2]
```diff
@@ -22,14 +22,14 @@ Pod::Spec.new do |s|
...
+  s.dependency 'GoogleMLKit/TextRecognitionChinese', '4.0.0'
   # To recognize Devanagari script
-  s.dependency 'GoogleMLKit/TextRecognitionDevanagari', '2.6.0'
+  s.dependency 'GoogleMLKit/TextRecognitionDevanagari', '4.0.0'
   # To recognize Japanese script
-  s.dependency 'GoogleMLKit/TextRecognitionJapanese', '2.6.0'
+  s.dependency 'GoogleMLKit/TextRecognitionJapanese', '4.0.0'
   # To recognize Korean script
-  s.dependency 'GoogleMLKit/TextRecognitionKorean', '2.6.0'
+  s.dependency 'GoogleMLKit/TextRecognitionKorean', '4.0.0'
```
Comment (a7medev) [inline text-recognition/RNMLKitTextRecognition.podspec:33]: Let's extract the dependency updates part into another PR updating dependencies for all the packages as they need to be kept in-sync to not break apps that use multiple packages.

**Review by a7medev — CHANGES_REQUESTED** [review]

File: text-recognition/index.d.ts (hunk 1 above)
Comment (spetrey) [reply text-recognition/index.d.ts:75]: Totally onboard with this change. Will address tonight 👍

**Comment by spetrey** [comment]
@a7medev great feedback on the PR! Much appreciated. I addressed your comments and went a little further to do some additional cleanup. I also figured I would rename the `language` param to `script` [here](https://github.com/a7medev/react-native-ml-kit/pull/28/files#diff-e27ca47e4e001b5ab59852261c0b71b0a1b90eaba5f56c7a136edc762562678fR85) because it's based on Script names instead of Language names. See this table from ML Kit's text recognition v2 docs:

https://developers.google.com/ml-kit/vision/text-recognition/v2/languages#supported-langs

Let me know if you see anything else that needs addressing :)

File: text-recognition/index.d.ts [hunk 3]
```diff
@@ -57,14 +57,33 @@ export interface TextRecognitionResult {
   blocks: TextBlock[];
 }
 
+export enum TranslateRecognitionScript {
```
Comment (a7medev) [inline text-recognition/index.d.ts:60]: I think it should be `TextRecognitionScript`.

```suggestion
export enum TextRecognitionScript {
```

**Review by a7medev — CHANGES_REQUESTED** [review]

**Comment by a7medev** [comment]
@spetrey I think will also need to support for Android as well to keep both platforms consistent. What do you think?

**Comment by spetrey** [comment]
@a7medev apologies on the delay getting back to you. I do agree that supporting Android is important. Unfortunately, I have never worked on any Java projects before this, so I may be unable to contribute to supporting Android on this PR (in the short-term) 😞 

What're your thoughts on merging this PR, then creating an Issue to track adding support for `TextRecognition` on Android at a later date?

**Comment by a7medev** [comment]
@spetrey Sounds good. Thanks for your contribution! 🙏
I'll test this PR on iOS and merge it and will work on another PR for Android.

**Comment by spetrey** [comment]
Thank you for taking the time to review! Let me know if you encounter any issues testing @a7medev, happy to resolve anything that comes up 👍

File: text-recognition/ios/TextRecognition.m [hunk 4]
```diff
@@ -75,15 +79,35 @@ - (NSDictionary*)blockToDict: (MLKTextBlock*)block {
...
     NSData *imageData = [NSData dataWithContentsOfURL:_url];
     UIImage *image = [UIImage imageWithData:imageData];
     MLKVisionImage *visionImage = [[MLKVisionImage alloc] initWithImage:image];
-    MLKTextRecognizer *textRecognizer = [MLKTextRecognizer textRecognizer];
+
+    // text recognizer options based on the script params
+    MLKTextRecognizerOptions *options = nil;
+
+    // if the language param isn't specified, we can assume the user requirement is Latin text recognition
+    if (script == nil) {
```
Comment (a7medev) [inline text-recognition/ios/TextRecognition.m:95]:
```suggestion
    if (script == nil || [script isEqualToString:@"Latin"]) {
```

**Review by a7medev — CHANGES_REQUESTED** [review]
Calling `TextRecognition.recognize(imagePath)` without the `script` parameter doesn't work and throws this error: `TextRecognition.recognize was called with 1 arguments but expected 2 arguments.`

We need to make sure this feature is backward-compatible by providing the default value of `"Latin"` on iOS and not passing the `script` parameter on Android which will require us to make a wrapper method around the `TextRecognition.recognize` native method but without changing the API to not break the current version of the library. I'll handle that case and merge the PR once ready.

Also, there's another change for making the `"Latin"` enum value work, please check it.

Thanks so much for your contribution! 🙏🏼

File: text-recognition/ios/TextRecognition.m (hunk 4 above)
Comment (spetrey) [reply text-recognition/ios/TextRecognition.m:95]: Ah, good suggestion!

**Comment by spetrey** [comment]
Great feedback @a7medev! Thanks for double-checking my contribution and ensuring backward compat on Android 🙏

**Review by a7medev — APPROVED** [review]
Looks great, let's ship it in the next release! 🚀

**Comment by cardanoeconews** [comment]
Hi @a7medev and @spetrey , Just checking in to see if Android support for non-latin has been added or if you've started on that MR @a7medev ? It's something needed before I can ship my product as there are a ton of android users in South Korea. Would love any help, happy to sponsor!

**Comment by a7medev** [comment]
Hi @cardanoeconews! Unfortunately, I haven't had much time to work on adding Android support lately, will work on it once I have some time. 🙏🏼 Happy to know you are using it in production!

I don't have GitHub Sponsors set up at the moment, but I'm working on it—having some support would actually allow me to spend more time working on these projects and delivering more features.

**Comment by a7medev** [comment]
Wow, that was fast! 😅 I've set up GitHub Sponsors, you can sponsor my work on React Native ML Kit and other packages through [my Sponsors page](https://github.com/sponsors/a7medev)!

**Comment by a7medev** [comment]
Quick update! Non-Latin script Android support just landed in `@react-native-ml-kit/text-recognition` v1.5.0 🚀
